packages/protocol/contracts/identity/FederatedAttestations.sol

```diff
@@ -174,10 +251,122 @@ contract FederatedAttestations is
         trimmedIssuedOns[i] = issuedOns[i];
         trimmedSigners[i] = signers[i];
       }
-      return (trimmedAccounts, trimmedIssuedOns, trimmedSigners);
+      return (countsPerIssuer, trimmedAccounts, trimmedIssuedOns, trimmedSigners);
     } else {
-      return (accounts, issuedOns, signers);
+      return (countsPerIssuer, accounts, issuedOns, signers);
+    }
+  }
+
+  /**
+   * @notice Similar to lookupAttestations but returns all attestations for
+   *   an identifier produced by a list of issuers, regardless of whether
+   *   signers have been revoked or not
+   * @param identifier Hash of the identifier
+   * @param trustedIssuers Array of n issuers whose attestations will be included
+   * @return [0] Array of number of attestations returned per issuer
+   * @return [1 - 3] for m (== sum([0])) found attestations:
+   *         [
+   *           Array of m accounts,
+   *           Array of m issuedOns,
+   *           Array of m signers
+   *         ]; index corresponds to the same attestation
+   * @dev Adds attestation info to the arrays in order of provided trustedIssuers
+   * @dev Expectation that only one attestation exists per (identifier, issuer, account)
+   */
+  function lookupAllAttestations(bytes32 identifier, address[] calldata trustedIssuers)
```

Comment:
In this case, the precomputation step is far more efficient since we don't need to check revocation status (storage access) for every attestation. I know it's a bit lopsided as is, but I think there's a clearer use case for getting all attestations (instead of "top 1") -- i.e. so you can delete revoked attestations, but less for the maxAttestations case. I think there's a use case for both for unrevoked attestations. Does that make sense? As mentioned in the other comment, I think we can additionally add a `lookupAllUnrevoked` that wraps the getMax & lookupAll calls to allow this to happen atomically.